modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -714,6 +717,35 @@ def set_cancellable_false(appointment)
         appointment[:cancellable] = false
       end
 
+      def set_modality(appointment)
+        modality = nil
+        if appointment[:service_type] == 'covid'
+          modality = 'vaInPersonVaccine'
+        elsif appointment.dig(:service_category, 0, :text) == 'COMPENSATION & PENSION'
+          modality = 'claimExamAppointment'
+        elsif appointment[:kind] == 'clinic'
+          modality = 'vaInPerson'
+        elsif appointment[:kind] == 'telehealth'
+          modality = telehealth_modality(appointment)
+        elsif appointment[:kind] == 'phone'
+          modality = 'vaPhone'
+        elsif appointment[:kind] == 'cc'
+          modality = 'communityCare'
+        end
+
+        appointment[:modality] = modality
+      end
```

Comment:
You could probably use a case/when since Ruby 2.7 added pattern matching. I also added a nil check:

```suggestion
      def set_modality(appointment)
        raise ArgumentError, 'Appointment cannot be nil' if appointment.nil?

        appointment[:modality] = case
        when appointment[:service_type] == 'covid'
          'vaInPersonVaccine'
        when appointment.dig(:service_category, 0, :text) == 'COMPENSATION & PENSION'
          'claimExamAppointment'
        when appointment[:kind] == 'clinic'
          'vaInPerson'
        when appointment[:kind] == 'telehealth'
          telehealth_modality(appointment)
        when appointment[:kind] == 'phone'
          'vaPhone'
        when appointment[:kind] == 'cc'
          'communityCare'
        else
          raise ArgumentError, "Unknown appointment type."
        end
      end
```